src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -0,0 +1,373 @@
+            case 'sqlite':
+                $sql = "CREATE TABLE $this->table ($this->idCol TEXT NOT NULL PRIMARY KEY, $this->tokenCol TEXT NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            case 'pgsql':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR(64) NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            case 'oci':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR2(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR2(64) NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            case 'sqlsrv':
+                $sql = "CREATE TABLE $this->table ($this->idCol VARCHAR(64) NOT NULL PRIMARY KEY, $this->tokenCol VARCHAR(64) NOT NULL, $this->expirationCol INTEGER)";
+                break;
+            default:
+                throw new \DomainException(sprintf('Creating the lock table is currently not implemented for PDO driver "%s".', $driver));
+        }
+
+        $conn->exec($sql);
+    }
+
+    /**
+     * Cleanup the table by removing all expired locks
+     */
+    private function prune(): void
+    {
+        $sql = "DELETE FROM $this->table WHERE $this->expirationCol <= {$this->getCurrentTimestampStatment()}";
+
+        $stmt = $this->getConnection()->prepare($sql);
+
+        $stmt->execute();
+    }
+
+    private function getDriver(): string
+    {
+        if (null !== $this->driver) {
+            return $this->driver;
+        }
+
+        $con = $this->getConnection();
+        if ($con instanceof \PDO) {
+            $this->driver = $con->getAttribute(\PDO::ATTR_DRIVER_NAME);
+        } else {
+            switch ($this->driver = $con->getDriver()->getName()) {
+                case 'mysqli':
+                case 'pdo_mysql':
+                case 'drizzle_pdo_mysql':
+                    $this->driver = 'mysql';
+                    break;
+                case 'pdo_sqlite':
+                    $this->driver = 'sqlite';
+                    break;
+                case 'pdo_pgsql':
+                    $this->driver = 'pgsql';
+                    break;
+                case 'oci8':
+                case 'pdo_oracle':
+                    $this->driver = 'oci';
+                    break;
+                case 'pdo_sqlsrv':
+                    $this->driver = 'sqlsrv';
+                    break;
+            }
+        }
+
+        return $this->driver;
+    }
+
+
+    private function getCurrentTimestampStatment(): string
+    {
+        switch ($this->getDriver()) {
+            case 'mysql':
+                return 'UNIX_TIMESTAMP()';
+            case 'sqlite':
+                return 'strftime(\'%s\',\'now\')';
+            case 'pgsql':
+                return 'CAST(EXTRACT(epoch FROM NOW()) AS INT)';
+            case 'oci':
+                return '(sysdate - to_date(\'1970-01-01 00:00:00\', \'YYYY-MM-DD HH24:MI:SS\')) * 86400';
```

Comment:
I'm not able to test OCI nor SQL server. This is theorical code found on internet..

I could fallback on php `time()` but this method is more sensible to clockDrift